examples/Remarketing/UpdateAudienceTargetRestriction.php

```diff
@@ -0,0 +1,241 @@
+
+                    // Adds an AUDIENCE target restriction with bid_only set to true to the
+                    // targeting setting object. This has the effect of setting the AUDIENCE
+                    // target restriction to "Observation".
+                    // For more details about the targeting setting, visit
+                    // https://support.google.com/google-ads/answer/7365594.
+                    $targetRestrictions[] = new TargetRestriction([
+                        'targeting_dimension' => TargetingDimension::AUDIENCE,
+                        'bid_only' => true
+                    ]);
+                }
+            }
+        }
+
+        // Only updates the TargetingSetting on the ad group if there is an AUDIENCE
+        // TargetRestriction with bid_only set to false.
+        if ($shouldUpdateTargetingSetting) {
+            self::updateTargetingSetting(
+                $googleAdsClient,
+                $customerId,
+                $adGroupId,
+                new TargetingSetting([
+                    'target_restrictions' => $targetRestrictions
+                ])
+            );
+        } else {
+            print "No target restrictions to update." . PHP_EOL;
+        }
+    }
+
+    /**
+     * Updates the TargetingSetting of an ad group.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ID of the ad group to update
+     * @param TargetingSetting $targetingSetting the updated targeting setting
+     */
+    public static function updateTargetingSetting(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId,
+        TargetingSetting $targetingSetting
+    ) {
+        // Creates an ad group object with the proper resource name and updated targeting setting.
+        $adGroup = new AdGroup([
+            'resource_name' => ResourceNames::forAdGroup($customerId, $adGroupId),
+            'targeting_setting' => $targetingSetting
+        ]);
+
+        // Constructs an operation that will update the ad group with the specified resource name,
+        // using the FieldMasks utility to derive the update mask. This mask tells the Google Ads
+        // API which attributes of the ad group you want to change.
+        $adGroupOperation = new AdGroupOperation();
+        $adGroupOperation->setUpdate($adGroup);
+        $adGroupOperation->setUpdateMask(FieldMasks::allSetFieldsOf($adGroup));
+
+        // Issues a mutate request to update the ad group.
+        $adGroupServiceClient = $googleAdsClient->getAdGroupServiceClient();
+        $response = $adGroupServiceClient->mutateAdGroups(
+            $customerId,
+            [$adGroupOperation]
+        );
+
+        // Prints the resource name of the updated ad group.
+        printf(
+            "Updated targeting setting of ad group with resourceName '%s'; set the AUDIENCE " .
```

Comment:
Nit: "with resource name"

Reply:
Agreed, I opened https://github.com/googleads/google-ads-java/pull/347 to fix the golden as well.

Reply:
FYI, Java may be a bit different. We stick to common English words for long, but other repos seem to have a mix use of both terms.